### ISSUE TITLE
Create EthereumAddress interface and class

### ISSUE THREAD

**Issue by brunobar79** [MetaMask/gaba#25]
By having an interface EthereumAddress we can forget about all the type related issues (string vs hex, etc).

If we have an EthereumAddress obj also we could call toChecksumAddress inside of the constructor and have some utility function like toShortString (which could return `0x1234...1234`), etc.

cc: @danfinlay

**Comment by bitpshr** [comment]
Agreed. We should use this type both internally as soon as possible and eventually externally once all consuming applications are also using TypeScript.
